### PR TITLE
LPS-142443 Add labels element to nodes

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**
Story: [LPS-139387](https://issues.liferay.com/browse/LPS-139387)

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/dto/v1_0/util/NodeUtil.java [hunk 1]
```diff
@@ -15,25 +15,43 @@
...
-		WorkflowNode.Type type) {
+		Language language, Locale locale, WorkflowNode workflowNode) {
 
 		Node node = new Node();
 
-		node.setLabel(language.get(resourceBundle, name));
-		node.setName(name);
-		node.setType(Node.Type.create(type.name()));
+		if (MapUtil.isEmpty(workflowNode.getLabelMap())) {
+			node.setLabel(
```
Comment (rafaprax) [inline modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/dto/v1_0/util/NodeUtil.java:38]: Hey @MarinhoFeliphe, please use the set methods that accept an UnsafeSupplier instead. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:workflow-metrics

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:workflow-metrics

**Comment by MarinhoFeliphe** [comment]
ci:test:workflow

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:workflow

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoNodeLocalServiceImpl.java [hunk 2]
```diff
@@ -68,6 +68,7 @@ public KaleoNode addKaleoNode(
 		kaleoNode.setModifiedDate(date);
 		kaleoNode.setKaleoDefinitionId(kaleoDefinitionId);
 		kaleoNode.setKaleoDefinitionVersionId(kaleoDefinitionVersionId);
+		kaleoNode.setLabelMap(node.getLabelMap());
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoNodeLocalServiceImpl.java:71]: after name

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/dto/v1_0/util/NodeUtil.java [hunk 3]
```diff
@@ -15,25 +15,47 @@
...
+						ResourceBundleUtil.getModuleAndPortalResourceBundle(
+							locale, WorkflowDefinitionResourceImpl.class),
+						workflowNode.getName());
+				}
+
+				Map<Locale, String> labelMap = workflowNode.getLabelMap();
+
+				return labelMap.get(locale);
+			});
+
```
Comment (MarinhoFeliphe) [inline modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/dto/v1_0/util/NodeUtil.java:50]: remove blank lines

File: modules/apps/portal-workflow/portal-workflow-kaleo-definition-impl/src/main/java/com/liferay/portal/workflow/kaleo/definition/internal/export/BaseNodeExporter.java [hunk 4]
```diff
@@ -237,6 +242,24 @@ else if (assignmentType.equals(AssignmentType.USER)) {
...
+		if (MapUtil.isEmpty(node.getLabelMap())) {
+			return;
+		}
+
+		Map<Locale, String> labelMap = node.getLabelMap();
+
+		Element labelsElement = element.addElement("labels");
+
+		for (Map.Entry<Locale, String> entry : labelMap.entrySet()) {
+			Element label = labelsElement.addElement("label");
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-kaleo-definition-impl/src/main/java/com/liferay/portal/workflow/kaleo/definition/internal/export/BaseNodeExporter.java:255]: labelElement

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
Just started reviewing :)

**Pull request closed without merging**